### PR TITLE
Kian branch

### PR DESCRIPTION
**Pull request by KianShah**

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -22,141 +20,4 @@ router.get("/test", (req, res) => {
   res.sendStatus(200);
 });
 
-
-router.post('/submitJob', (req, res) => {
```
Comment (nikkimoteva) [inline server/routes/api.js:26]: Where did this function get moved to?

File: server/routes/api.js [hunk 2]
```diff
@@ -22,141 +20,4 @@ router.get("/test", (req, res) => {
...
-    .then(msg => {
-      res.status(200);
-      res.send({ borg_stdout: msg, fileHash: file_name });
-    })
-    .catch(err => {
-      errorHandler(err, res);
-    });
-});
-
-router.post('/pipeline', (req, res) => {
```
Comment (nikkimoteva) [inline server/routes/api.js:94]: Where did this function get moved to?

File: server/routes/JobRoutes.js [hunk 3]
```diff
@@ -76,44 +79,68 @@ router.post("/submitTrainJob", (req, res) => {
...
+    .then(job => {
+      // if (!(job.headers.includes(job.target_name))) { // gotta push the target column into the file!
+      //   const a = job.headers;
+      //   a.push(job.target_name);
+      //   data[0] = testColumns;
+      //   const newFile = csv.fromArrays(data);
+      //
+      const folder_path = slurm_command_dataset_path + job.fileHash;
+      const test_path = folder_path + '/' + test_file_name + '.csv';
+      const local_path = '/home/kians376/repos/cs455/unDecided/server/predictions/' + job.fileHash + '.csv';
```
Comment (robertmzm) [inline server/routes/JobRoutes.js:124]: Use relative path instead of absolute path

Comment (robertmzm) [inline server/routes/JobRoutes.js]: Use relative path instead of absolute path

File: server/routes/JobRoutes.js [hunk 4]
```diff
@@ -35,70 +40,107 @@ router.delete("/deleteJob", (req, res) => {
...
-  const maxJobTime = body.maxJobTime;
-  const targetColumnName = body.targetColumnName;
-  const header = body.header;
+  console.log(req.body);
+  const {id_token, jobName, maxJobTime, targetColumnName, dataset, header} = req.body;
   const targetColumn = header.indexOf(targetColumnName);
-  const dataset = body.dataset;
+  const file_name = uuidv4();
+  const train_path = borg_dataset_directory + file_name + "/train.csv";
+  const local_path = "/home/kians376/repos/cs455/unDecided/server/training/" + file_name + ".csv";
```
Comment (nikkimoteva) [inline server/routes/JobRoutes.js]: Absolute path won’t work on other machines. Maybe try ‘../training’ ?

**Review by nikkimoteva — CHANGES_REQUESTED** [review]

**Review by robertmzm — APPROVED** [review]
tested

File: src/common/Managers/EndpointManager.js [hunk 5]
```diff
@@ -34,13 +34,13 @@ export function submitJob(id_token, jobName, maxJobTime, targetColumnName, datas
...
-    dataset: "", // TODO: Fix this so we can send dataset to the server properly
+    dataset: dataset,
     header
   });
 }
 
-export function submitPrediction(id_token, predictionName, jobID, ) {
-  return axios.post(`${jobRoot}/submitPrediction`, {id_token, predictionName, jobID});
+export function submitPrediction(id_token, predictionName, jobID, dataset) {
+  return axios.post(`${jobRoot}/submitPrediction`, {id_token, predictionName, jobID: '6104e8a20ed9b2030ef05fd3', dataset});
```
Comment (robertmzm) [inline src/common/Managers/EndpointManager.js]: I believe this fixed jobid is given just for testing purpose only

**Review by robertmzm — CHANGES_REQUESTED** [review]

File: server/routes/api.js (hunk 1 above)
Comment (KianShah) [reply server/routes/api.js:26]: submitTrainJob in JobRoutes

File: server/routes/api.js (hunk 2 above)
Comment (KianShah) [reply server/routes/api.js:94]: submitPrediction in jobRoutes

**Review by nikkimoteva — APPROVED** [review]

**Review by robertmzm — APPROVED** [review]

**Pull request merged**